[JBPM-9626] - Create test case for Long running queries at Kie server startup

afalhambra opened the pull request:
**[JBPM-9626](https://issues.redhat.com/browse/JBPM-9626)**: Create test case for Long running queries at Kie server startup

jstastny-cz commented on jbpm-wb-kie-server/jbpm-wb-kie-server-backend/src/test/java/org/jbpm/workbench/ks/integration/KieServerQueryDefinitionLoaderTest.java:
```diff
@@ -0,0 +1,79 @@
+    }
+
+    @Test
+    public void testJbpmProcessInstancesWithVariables() {
```
Please add such test for all the named queries similarly to what you did for `jbpmProcessInstancesWithVariables` here

afalhambra replied:
done

jstastny-cz commented on jbpm-wb-kie-server/jbpm-wb-kie-server-backend/src/test/java/org/jbpm/workbench/ks/integration/KieServerQueryDefinitionLoaderTest.java:
```diff
@@ -0,0 +1,79 @@
+
+    @Test
+    public void testJbpmProcessInstancesWithVariables() {
+        QueryDefinition query = QueryDefinition.builder()
```
```suggestion
        QueryDefinition expectedQuery = QueryDefinition.builder()
```

afalhambra replied:
done

jstastny-cz commented on jbpm-wb-kie-server/jbpm-wb-kie-server-backend/src/test/java/org/jbpm/workbench/ks/integration/KieServerQueryDefinitionLoaderTest.java:
```diff
@@ -0,0 +1,79 @@
+        testQueryDefinitionLoaded(query);
+    }
+
+    private void testQueryDefinitionLoaded(QueryDefinition queryDefinition) {
```
```suggestion
    private void testQueryDefinitionLoaded(QueryDefinition expectedQueryDefinition) {
```

afalhambra replied:
done

jstastny-cz commented on jbpm-wb-kie-server/jbpm-wb-kie-server-backend/src/test/java/org/jbpm/workbench/ks/integration/KieServerQueryDefinitionLoaderTest.java:
```diff
@@ -0,0 +1,79 @@
+        }).when(event).fire(any(QueryDefinitionLoaded.class));
+        this.kieServerQueryDefinitionLoader.init();
+
+        assertTrue(receivedEvents.size() > 0);
```
Is a more descriptive assertion available, e.g. using hamcrest?

afalhambra replied:
Just added some description to the assert. Let me know if that's ok for you, otherwise I can use hamcrest, but it will require a change in the pom.xml as this dep is not included there

gmunozfe reviewed (approved):
Looks good to me, gorgeous work @afalhambra !
Perhaps you can test also some rainy scenario (non-existing named query) but up to you.

afalhambra commented:
> Looks good to me, gorgeous work @afalhambra !
> Perhaps you can test also some rainy scenario (non-existing named query) but up to you.

Added, thanks @gmunozfe

jstastny-cz reviewed (approved):
LGTM. Thanks.

gmunozfe commented:
jenkins retest this

MarianMacik reviewed (commented):
What if we added one more test case which will test the number of queries fetched?
By doing this we will have a complete coverage in case:
* Somebody changes the expression
* Somebody deletes the whole query
* Somebody adds a new query

In all these scenarios a test suite should fail and author should properly add/change tests so they match. We could also write a JavaDoc for this test why it is important etc.

What do you think?

afalhambra commented:
> What if we added one more test case which will test the number of queries fetched?
> By doing this we will have a complete coverage in case:
> 
> * Somebody changes the expression
> * Somebody deletes the whole query
> * Somebody adds a new query
> 
> In all these scenarios a test suite should fail and author should properly add/change tests so they match. We could also write a JavaDoc for this test why it is important etc.
> 
> What do you think?

Agree. Added Javadoc along with a new test case for checking the number of query definitions loaded.
